pkg/api/v1/types.go

```diff
@@ -437,8 +440,21 @@ const (
 
 	// MountOptionAnnotation defines mount option annotation used in PVs
 	MountOptionAnnotation = "volume.beta.kubernetes.io/mount-options"
+
+	// StorageTopologyConstraint defines topology constraints of a PV
+	// Value is type []TopologyConstraint
+	AlphaStorageTopologyConstraintsAnnotation = "volume.alpha.kubernetes.io/topology-constraints"
 )
 
+// TopologyConstraint contains label information that the resource is constrained to
+// A label key can have multiple values
+type TopologyConstraint struct {
+	// Key of the label
```

Comment:
There's another debate going on about whether a NodeName label on a Node object makes sense.  The other pod scheduling features all use the hostname label, so if we added a nodename label, that could potentially also add some consistency confusion when using local PVs with pod affinity.  I would rather handle it separately than block this change.